src/EPR.Calculator.Frontend/Controllers/ParameterUploadFileProcessingController.cs

```diff
@@ -31,7 +31,7 @@ public IActionResult Index([FromBody] List<SchemeParameterTemplateValue> schemeP
                     throw new ArgumentNullException(parameterSettingsApi, "ParameterSettingsApi is null. Check the configuration settings for default parameters");
                 }
 
-                this.FileName = this.TempData.Peek("FileName").ToString();
+                this.FileName = this.HttpContext.Session.GetString("FileName");
```

Comment:
Please create constants for session variables
Also, you need to clean up all the values in try/catch block. Thanks